pytorch_lightning/trainer/supporters.py

```diff
@@ -120,23 +130,126 @@ def mean(self):
 
 
 class PredictionCollection(object):
-    def __init__(self, global_rank: int, world_size: int):
+    def __init__(self, global_rank: int, world_size: int, all_gather: Callable):
         self.global_rank = global_rank
         self.world_size = world_size
-        self.predictions = {}
-        self.num_predictions = 0
+        self.all_gather = all_gather
+        self._legacy_predictions = {}
+        self._predictions = {stage: {} for stage in LoggerStages}
+
+    @property
+    def predictions(self):
+        return self._predictions[self.current_stage]
+
+    @property
+    def should_all_gather(self):
+        return torch.distributed.is_available() or not torch.distributed.is_initialized() and self.world_size > 1
```

Comment:
this looks as static func, why as a property?

Reply:
Can't be a static method due to `self.world_size > 1`